File: Todo-App/backend-fastapi/test_gemini.py
import asyncio
import os
from dotenv import load_dotenv
from src.utils.gemini_nlp_processor import gemini_nlp_processor

# Load environment variables
load_dotenv()

async def test_gemini_integration():
    print("Testing Gemini API integration...")

    try:
        # Test the NLP processor
        print("\nTesting Gemini NLP Processor...")

        test_inputs = [
            "Add a task to buy groceries",
            "Complete task 1",
            "Delete task 2",
            "Show me my tasks for today",
            "Find tasks about meeting",
            "Edit task 3 to update the description"
        ]

        for input_text in test_inputs:
            print(f"\nProcessing: '{input_text}'")
            result = await gemini_nlp_processor.process_input(input_text)
            print(f"Result: {result}")

        print("\n✅ Gemini integration test completed successfully!")

    except Exception as e:
        print(f"❌ Error during Gemini integration test: {str(e)}")

if __name__ == "__main__":
    asyncio.run(test_gemini_integration())

File: Todo-App/backend/test-gemini.js
// Test script to verify Gemini API integration
require('dotenv').config();
const { GoogleGenerativeAI } = require("@google/generative-ai");

// Initialize the Gemini API client
const genAI = new GoogleGenerativeAI(process.env.GEMINI_API_KEY);
const model = genAI.getGenerativeModel({ model: "gemini-pro" });

async function testGeminiAPI() {
  try {
    console.log("Testing Gemini API connection...");

    const prompt = "Say hello and tell me you're a task management assistant.";
    const result = await model.generateContent(prompt);
    const response = await result.response;
    const text = response.text();

    console.log("Gemini API Response:", text);
    console.log("✅ Gemini API is working correctly!");

    // Test the NLP processor
    console.log("\nTesting Gemini NLP Processor...");
    const geminiNlpProcessor = require('./utils/geminiNlpProcessor');

    const testInputs = [
      "Add a task to buy groceries",
      "Complete task 1",
      "Delete task 2",
      "Show me my tasks for today"
    ];

    for (const input of testInputs) {
      console.log(`\nProcessing: "${input}"`);
      const result = await geminiNlpProcessor.processInput(input);
      console.log("Result:", JSON.stringify(result, null, 2));
    }

  } catch (error) {
    console.error("❌ Error testing Gemini API:", error.message);
  }
}

// Run the test
testGeminiAPI();